tests/Feature/RegisterUserTest.php

```diff
@@ -23,328 +25,572 @@ public function setUp(): void
 
-        $this->createAccount(['phone' => '(650) 253-0000']) // Google California number
-        ->assertSessionDoesntHaveErrors();
+        $this->fillForm(Arr::only(
+            $userInvitation->toArray(),
+            ['first_name', 'last_name', 'class_course', 'class_year']
+        ))
+            ->call('run');
 
-        $this->assertEquals('US', Auth::user()->phone->getCountry());
+        $this->assertDatabaseMissing('user_invitations', ['id' => $userInvitation->id]);
+    }
 
-        Auth::logout();
+    /** @test */
+    public function testUserLoggedInUponRegistration()
+    {
+        $this->fillForm()->call('run');
 
-        $this->createAccount(['phone' => '0370 010 0222']) // BBC London number
-        ->assertSessionHasErrors('phone');
+        $this->assertAuthenticated();
     }
 
-    /**
-     * Submits a post request to create an account.
-     *
-     * @param  array  $overrides
-     * @param  bool  $invited
-     * @return \Illuminate\Testing\TestResponse
-     */
-    public function createAccount(array $overrides = [], bool $invited = true)
-    {
-        $userData = array_merge(
-            Arr::only(make(User::class)->getAttributes(), [
-                'first_name',
-                'last_name',
-                'email',
-                'class_course',
-                'class_year',
-                'gender',
-                'birthdate',
-                'phone',
-            ]),
-            ['password' => 'HugoWasHere', 'password_confirmation' => 'HugoWasHere']
+    /** @test */
+    public function testConfirmationEmailSentUponRegistration()
+    {
+        $this->fillForm()->call('run');
+
+        Notification::assertSentTo(
+            [User::latest()->first()], VerifyEmail::class
         );
+    }
+
+    /** @test */
+    public function testNewUserCanVerifyTheirEmail()
+    {
+        $this->fillForm()->call('run');
+
+        $user = Auth::user();
+        $this->flushSession();
+
+        $this->assertNull($user->email_verified_at);
 
-        if ($invited) {
-            $this->inviteUser($userData, $overrides);
-        }
+        $this->get((new VerifyEmail)->toMail($user)->actionUrl)
+            ->assertRedirect(route('home'));
 
-        return $this->post(route('register'), array_merge(
-            $userData,
-            $overrides
-        ));
+        $this->assertNotNull($user->email_verified_at);
+    }
+
+    /** @test */
+    public function testPhoneCountryCanBeAutoDetectedFromCountryCode()
+    {
+        $this->fillForm(['phone' => '+1 (202) 456-1414'])->call('run'); // White House
+
+        $this->assertEquals('US', Auth::user()->phone->getCountry());
     }
 
     /**
-     * Creates a UserInvitation with the given data.
+     * TODO: Find a way to force the server variable change.
+     * @test
+     * @group external-api
+     * @group geoip-api
+     */
+//    public function testPhoneWithoutCountryCodeFallsbackToFranceThenGeoIpBased()
```

Comment:
We'll need to find an appropriate way to test this functionality.
A PR to the Livewire may be required.

Reply:
Issue opened for tracking: #47